core/store/migrate/migrations/0115_migrate_upkeep_id_to_big_int.sql

```diff
@@ -0,0 +1,48 @@
+-- +goose Up
+-- +goose StatementBegin
+
+ALTER TABLE upkeep_registrations ALTER COLUMN upkeep_id TYPE numeric(78,0);
+
+CREATE OR REPLACE FUNCTION uint256_to_bit(NUMERIC)
+  RETURNS BIT VARYING AS $$
+DECLARE
+  num ALIAS FOR $1;
+  -- 1 + largest positive BIGINT --
+  max_bigint NUMERIC := '9223372036854775808' :: NUMERIC(19, 0);
+  result BIT VARYING;
+BEGIN
+  ASSERT num <= 115792089237316195423570985008687907853269984665640564039457584007913129639935, "value larger than max uint256";
+  WITH
+      chunks (exponent, chunk) AS (
+        SELECT
+          exponent,
+          floor((num / (max_bigint ^ exponent) :: NUMERIC(256, 20)) % max_bigint) :: BIGINT
+        FROM generate_series(0, 5) exponent
```

Comment:
This went over my head. Could you add some comments explaining what's happening here